openshift-serverless-clients.spec

```diff
@@ -0,0 +1,98 @@
+#debuginfo not supported with Go
+%global debug_package %{nil}
+%global package_name openshift-serverless-clients
+%global product_name OpenShift Serverless
+%global golang_version 1.13
+%global kn_version 0.13.1
+%global kn_release 1
+%global kn_cli_version v%{kn_version}
+%global source_dir knative-client-%{kn_version}-%{kn_release}
+%global source_tar %{source_dir}.tar.gz
+
+Name:           %{package_name}
+Version:        %{kn_version}
+Release:        %{kn_release}%{?dist}
+Summary:        %{product_name} client kn CLI binary
+License:        ASL 2.0
+URL:            https://github.com/openshift/knative-client/tree/release-%{kn_cli_version}
+
+ExclusiveArch:  x86_64
+
+Source0:        %{source_tar}
+BuildRequires:  gcc
+BuildRequires:  golang >= %{golang_version}
+Provides:       %{package_name}
+Obsoletes:      %{package_name} < %{kn_version}
+
+%description
+Client kn provides developer experience to work with Knative Serving APIs.
+
+%prep
+%setup -q -n %{source_dir}
+
+%build
+TAG=%{kn_cli_version} make build-cross
+
+%install
+mkdir -p %{buildroot}/%{_bindir}
+install -m 0755 kn-linux-amd64 %{buildroot}/%{_bindir}/kn
+
+install -d %{buildroot}%{_datadir}/%{name}-redistributable/{linux,macos,windows}
+install -p -m 755 kn-linux-amd64 %{buildroot}%{_datadir}/%{name}-redistributable/linux/kn-linux-amd64
+install -p -m 755 kn-darwin-amd64 %{buildroot}/%{_datadir}/%{name}-redistributable/macos/kn-darwin-amd64
+install -p -m 755 kn-windows-amd64.exe %{buildroot}/%{_datadir}/%{name}-redistributable/windows/kn-windows-amd64.exe
+
+%files
+%license LICENSE
+%{_bindir}/kn
+
+%package redistributable
+Summary:        %{product_name} client CLI binaries for Linux, macOS and Windows
+BuildRequires:  gcc
+BuildRequires:  golang >= %{golang_version}
+Provides:       %{package_name}-redistributable
```

Comment:
what's the diff between the normal package and the `-redistributable` package ?

Reply:
normal package for installing `kn` at RHEL machines.

`-redistributable` RPM to package cross-platform binaries together (line no 59 onwards)

excerpts:
```
%files redistributable
%license LICENSE
%dir %{_datadir}/%{name}-redistributable/linux/
%dir %{_datadir}/%{name}-redistributable/macos/
%dir %{_datadir}/%{name}-redistributable/windows/
%{_datadir}/%{name}-redistributable/linux/kn-linux-amd64
%{_datadir}/%{name}-redistributable/macos/kn-darwin-amd64
%{_datadir}/%{name}-redistributable/windows/kn-windows-amd64.exe
```

later these (brew built) binaries are extracted and used for signing+notarizing and sync at mirror.